src/Compatibility/Microsoft.DotNet.PackageValidation/ValidatePackage.cs

```diff
@@ -35,7 +33,7 @@ protected override void ExecuteCore()
 
             new CompatibleTfmValidator(NoWarn, null, RunApiCompat, logger).Validate(package);
             new CompatibleFrameworkInPackageValidator(NoWarn, null, logger).Validate(package);
-            if (BaselineValidation)
+            if (!string.IsNullOrEmpty(BaselinePackageTargetPath))
```

Comment:
Since we also allow the user to set this (as opposed to just using the version) we should validate that the path actually exists before we call CreatePackage. I'm fine with doing that check either here or in the targets file, but we should do it at some point in order to provide diagnostics in case the path doesn't exist

Reply:
Nuget parser will do that for us while loading it and throw that the package doesnot exist.

Reply:
https://github.com/dotnet/sdk/pull/17880#discussion_r639905957

Reply:
Have we tested how that looks for the user? If it is just an exception from the nuget parser with a callstack, then I'd argue that is not a great experience and instead opt to catch that case early and provide an error message with a human-readable message telling them what is wrong. If NuGet parser already throws a message instead of a callstack, then I'm fine.

Reply:
it returns with the call stack. personally i prefer the call stack as it gives some more info.

Reply:
I don't have a strong opinion here but avoiding an unnecessary read operation that will performed soon after sounds better IMO. I'll defer to @ericstj in this case.

Reply:
In the past in similar areas, we have opted to use detailed error messages as opposed to call stacks as most of the time that includes noise which is not super actionable and might confuse the user where the real thing we want to point out is just "The path that you passed in for us to check against the baseline doesn't exist".

If you feel strongly about keeping the callstack I'm fine, but in general I think it is better to have a human-readable message with less noise as that tends to be easier to take an action on.

Reply:
>I don't have a strong opinion here but avoiding an unnecessary read operation that will performed soon after sounds better IMO.

If what we want is to avoid the read, then we could always just add a try- catch block on the call to CreatePackage, and catch the exception and provide a better message in case the file is not found.